app/domain/authentication/authn_jwt/restriction_validation/create_constraints.rb

```diff
@@ -79,8 +77,14 @@ def add_required_constraint
           @constraints.append(required_constraint)
         end
 
+        def non_permitted_constraint
+          @non_permitted_constraint ||= @non_permitted_constraint_class.new(
+            non_permitted: @base_non_permitted_annotations + mapping_claims.keys
```

Comment:
```suggestion
            non_permitted: @base_non_permitted_annotations + mapping_claims.values
```
Keys are annotations and values are claim names: `brach:ref`.

Reply:
But i invert it

Reply:
![image](https://user-images.githubusercontent.com/73157235/126986639-966f02c7-5a16-4973-aa8d-b67e2c8e836e.png)